ocw_data_parser/ocw_data_parser.py

```diff
@@ -367,12 +384,43 @@ def generate_static_site(self):
             self.destination_dir + 'master/master.json',
             self.destination_dir + 'output/')
 
-    def export_master_json(self):
+    def export_master_json(self, s3_links=False):
+        if s3_links:
+            if not self.s3_bucket_name:
+                log.error("Please set your s3 bucket name")
+                return
+            bucket_base_url = f"https://{self.s3_bucket_name}.s3.amazonaws.com/{self.master_json['short_url']}/"
+            for p in self.compose_pages():
+                filename, html = htmlify(p)
+                if filename and html:
+                    update_file_location(
+                        self.master_json, bucket_base_url + filename, safe_get(p, "uid"))
+            for file in self.media_jsons:
+                uid = safe_get(file, "_uid")
+                filename = uid + "_" + safe_get(file, "id")
+                update_file_location(
+                    self.master_json, bucket_base_url + filename, uid)
+                if self.course_image_uid and uid == self.course_image_uid:
```

Comment:
Looks like a lot of this is copied from  `upload_all_media_to_s3`.  Can you extract it to a function?